torch/multiprocessing/spawn.py

```diff
@@ -131,9 +135,9 @@ def join(self, timeout=None):
                 process.terminate()
             process.join()
 
-        # There won't be an error on the queue if the process crashed.
+        # There won't be any content in the error file if the process crashed.
         failed_process = self.processes[error_index]
-        if self.error_queues[error_index].empty():
+        if os.stat(self.error_files[error_index]).st_size == 0:
             exitcode = self.processes[error_index].exitcode
             if exitcode < 0:
                 name = signal.Signals(-exitcode).name
```

Comment:
Kinda unrelated but this may raise a ValueError if the signal is a custom signal ?

Reply:
I'll send a PR for that.

Reply:
https://github.com/pytorch/pytorch/pull/114696